### PR TITLE
verify aodh

### PR DESCRIPTION
**Pull request by Xiaohua-Shen**
After verify aodh related alarming function in net-mitaka env, I found
some ceilometer and aodh configuration should be changed to support heat
autoscaling and auto recovery:
1. aodh lack of cafile for https
2. aodh's keystone_authtoken conf parameter is not correct
3. aodh need disable the rest_notifier_ssl_verify for notify heat. (same as old version of ursula)
4. ceilometer.conf's keystone-authtoken conf parameter is not correct
5.  add Lbaas's meter for autorecovery
6.  aodh's haproxy httpchk path is incorrect

### CODE REVIEW

File: roles/aodh/templates/etc/aodh/aodh.conf [hunk 1]
```diff
@@ -5,6 +5,8 @@ rpc_backend = rabbit
 log_dir = /var/log/aodh
 debug = {{ aodh.debug }}
 
+rest_notifier_ssl_verify = False
```
Comment (zfeldstein) [inline roles/aodh/templates/etc/aodh/aodh.conf]: We only want to do this if we are using a self signed certificate.

**Comment by zfeldstein** [comment]
Does python-memcached still need to be installed in the aodh venv?

**Comment by zfeldstein** [comment]
ok to test

**Comment by j2sol** [comment]
ok to test

File: roles/endpoints/defaults/main.yml [hunk 2]
```diff
@@ -238,6 +238,6 @@ endpoints:
       backend_api: 8043
       haproxy_api: 8042
     haproxy:
-      health_check: 'httpchk GET /healthcheck'
+      health_check: 'httpchk /'
```
Comment (j2sol) [inline roles/endpoints/defaults/main.yml]: Why did this get changed? we'd like to use the health check middleware. If this isn't working as is, I'd rather we added the middleware to aodh then to just hit `/`.

File: roles/aodh/templates/etc/aodh/aodh.conf [hunk 3]
```diff
@@ -35,17 +37,18 @@ connection = mysql+pymysql://aodh:{{ secrets.db_password }}@{{ endpoints.db }}/a
...
-user_domain_id = default
-project_name = service
-username = aodh
-password = {{ secrets.service_password }}
+admin_tenant_name = service
+admin_user = aodh
+admin_password = {{ secrets.service_password }}
 memcached_servers = {{ hostvars|ursula_memcache_hosts(groups, memcached.port) }}
+cafile = {{ aodh.cafile }}
+identity_uri = {{ endpoints.keystone.url.admin }}
```
Comment (j2sol) [inline roles/aodh/templates/etc/aodh/aodh.conf]: These read like old versions of arguments, is aodh using an older keystone library to handle these arguments?

File: roles/aodh/templates/etc/aodh/aodh.conf (hunk 1 above)
Comment (Xiaohua-Shen) [inline roles/aodh/templates/etc/aodh/aodh.conf]: without this, seems aodh-notifier (old as ceilometer-alarm-notifer) have error log about SSL when try to http callback to heat, and this config was used in old ceilometer.conf

[Fixed Heat Autoscaling By Making Ceilometer Alarms Not Verify SSL](https://github.com/blueboxgroup/ursula/commit/6e75494d6458ed2229fe4a6e3b2e7c9e6a3c09bc)

File: roles/aodh/templates/etc/aodh/aodh.conf (hunk 3 above)
Comment (Xiaohua-Shen) [inline roles/aodh/templates/etc/aodh/aodh.conf]: The aodh function verification is based on the net-mitaka env Niraj installed.
without these two parameter, aodh-api's https access cannot work. 

Please reference to mitaka version's aodh configuration document as below:
[Telemetry Alarming service configuration options](http://docs.openstack.org/mitaka/config-reference/telemetry/alarming_service_config_opts.html)

both cafile and identity_uri are mentioned, and not deprecated parameters.

File: roles/endpoints/defaults/main.yml (hunk 2 above)
Comment (Xiaohua-Shen) [inline roles/endpoints/defaults/main.yml]: aodh-api has no api path like /healthcheck.

please reference to [aodh v2 api](http://docs.openstack.org/developer/aodh/webapi/v2.html)
Also only swift use "/healthcheck", other like heat, keystone all use "/"

File: roles/aodh/templates/etc/aodh/aodh.conf (hunk 1 above)
Comment (Xiaohua-Shen) [inline roles/aodh/templates/etc/aodh/aodh.conf]: for the python-memcached, it's needed for aodh and ceilometer.  
it's not included in this change set. you can add it.

File: roles/endpoints/defaults/main.yml (hunk 2 above)
Comment (j2sol) [inline roles/endpoints/defaults/main.yml]: aodh can use the health check middleware provided by oslo. That can be a follow up PR though.

**Comment by j2sol** [comment]
retest

**Pull request merged**